### PR TITLE
Automatically enable Runner v2 for pipelines that use cross-language transforms.

### LINKED ISSUE #21787: [Feature Request]: Automatically upgrade Dataflow Java pipelines that use cross-language transforms to Runner v2
### What would you like to happen?

Such pipelines would not work with Dataflow Runner v1.

We already do this for Python SDK: https://github.com/apache/beam/blob/f131b2d20f91d8f427da047f28952834c8642747/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py#L418

### Issue Priority

Priority: 2

### Issue Component

Component: cross-language

### PR DESCRIPTION
**Pull request by chamikaramj**
…transforms

This fixes #21787 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: "addresses #123), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment "fixes #<ISSUE NUMBER>" instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @robertwb

**Comment by chamikaramj** [comment]
Run XVR_Dataflow PostCommit

**Comment by chamikaramj** [comment]
Dataflow CrossLanguageValidatesRunner PythonUsingJava Tests

**Comment by chamikaramj** [comment]
Dataflow CrossLanguageValidatesRunner JavaUsingPython Tests

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java [hunk 1]
```diff
@@ -1038,8 +1038,47 @@ private List<RunnerApi.ArtifactInformation> getDefaultArtifacts() {
...
+
   @Override
   public DataflowPipelineJob run(Pipeline pipeline) {
+    if (DataflowRunner.isMultiLanguagePipeline(pipeline)) {
+      List<String> experiments = options.getExperiments();
+      if (!experiments.contains("use_runner_v2")) {
+        LOG.info(
+            "Automatically enabling Dataflow Runner v2 since the pipeline used cross-language"
+                + "transforms");
+        experiments.add("use_runner_v2");
```
Comment (robertwb) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java:1079]: Do we have to add portable_job_submission as well, or is that not automatic for all Java pipelines?

**Review by robertwb — APPROVED** [review]

**Comment by robertwb** [comment]
Note that Jenkis is currently down, see https://issues.apache.org/jira/browse/INFRA-23358

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java (hunk 1 above)
Comment (chamikaramj) [reply runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java:1079]: That's automatic for all Java UW pipelines: https://github.com/apache/beam/blob/b8e2e85ab1fb37a2f89ed20d88730e591ea3bf7e/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L1055

**Comment by chamikaramj** [comment]
Retest this please

**Comment by chamikaramj** [comment]
Dataflow CrossLanguageValidatesRunner PythonUsingJava Tests

**Comment by chamikaramj** [comment]
Dataflow CrossLanguageValidatesRunner JavaUsingPython Tests

**Comment by chamikaramj** [comment]
Run Java PostCommit

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by chamikaramj** [comment]
(failures are unrelated)

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by chamikaramj** [comment]
Only failing test is "org.apache.beam.runners.flink.batch.ReshuffleTest.testEqualDistributionOnReshuffleAcrossMultipleStages" which is completed unrelated.

https://ci-beam.apache.org/job/beam_PreCommit_Java_Phrase/5235/

**Pull request merged**